File: backend/app/schemas/wallet.py
from pydantic import BaseModel


class AssetBalance(BaseModel):
    asset: str
    free: float
    locked: float = 0.0


class FuturesBalance(BaseModel):
    asset: str
    wallet_balance: float
    available_balance: float
    unrealized_profit: float = 0.0


class WalletSummary(BaseModel):
    credential_id: str
    credential_label: str
    spot: list[AssetBalance]
    futures: list[FuturesBalance]
    funding: list[AssetBalance]

File: frontend/src/api/wallet.ts
import { apiClient } from "./client";

export interface AssetBalance {
  asset: string;
  free: number;
  locked: number;
}

export interface FuturesBalance {
  asset: string;
  wallet_balance: number;
  available_balance: number;
  unrealized_profit: number;
}

export interface WalletSummary {
  credential_id: string;
  credential_label: string;
  spot: AssetBalance[];
  futures: FuturesBalance[];
  funding: AssetBalance[];
}

export async function fetchBalances(credentialId?: string): Promise<WalletSummary> {
  const { data } = await apiClient.get<WalletSummary>("/wallet/balances", {
    params: credentialId ? { credential_id: credentialId } : undefined,
  });
  return data;
}
